### PR TITLE
blog: jan/feb commit of the month

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @ellenkorbes, @nicks,

Please review the following commits I made in branch maiamcc/cotm-feb:

a890cd30785e76806fa8cc5df6a8d0e57741695f (2020-03-04 16:44:57 -0500)
blog: jan/feb commit of the month

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - tilt
+  - cotm
+keywords:
+  - tilt
+  - docker pull
+  - docker push
+  - transparency
+  - visibility
+---
+We missed our Commit of the Month post for January, so this post is doing double duty. We'll be discussing two exciting commits, which are two sides of the same coin: [e77be9fe](https://github.com/windmilleng/tilt/commit/e77be9fe2d97f5893ea99f131a1ef2ffdb4ec576) (Jan.) and [467f1913](https://github.com/windmilleng/tilt/commit/467f1913ea08b562e2e16b26d8d7e458b92c941b) (Feb.) **print `docker push` and `docker pull` progress (respectively) to the Web UI**, so you're not in the dark about where your development time is going.
```
Comment (nicks) [inline blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: I don't think this is accurate?

Here's the commit of the month post for January
https://blog.tilt.dev/2020/02/04/are-you-my-kustomize.html

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Tilt tries to make microservice development as fast as possible, but sometimes there's just stuff you've gotta wait around for. However, when you inevitably have to wait around for something, you should know exactly what what you're waiting for and how it's progressing (and that your process isn't just hung somewhere and FUBAR).
+
+These commits print `docker push / pull` progress output, updating lines in the Web UI in place as status meters creep forward. Users now have real-time visibility into what Docker is doing, instead of staring at a blank screen and hoping things are still working.
+
+![See docker push status in real time](/assets/images/jan-feb-2020-commit-of-the-month/push-output.gif)
+
+Plus, the functionality to update Web UI lines in place---which we first built for the Docker output use case---opens up tons of paths for improving visibility into what Tilt is doing at any given time.
```
Comment (nicks) [inline blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: One thing that might be fun to highlight here - when we were building this feature, @hyu and I spent a bunch of time studying the Docker CLI UX and how it's implemented. We really admire it! Han pointed out how they had put a lot of work into making the terminal experience great, and even making it responsive to your terminal width:

https://github.com/moby/moby/blob/master/pkg/jsonmessage/jsonmessage.go#L46

We wanted to carry over that experience to the Tilt Web UI

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+keywords:
+  - tilt
+  - docker pull
+  - docker push
+  - transparency
+  - visibility
+---
+We missed our Commit of the Month post for January, so this post is doing double duty. We'll be discussing two exciting commits, which are two sides of the same coin: [e77be9fe](https://github.com/windmilleng/tilt/commit/e77be9fe2d97f5893ea99f131a1ef2ffdb4ec576) (Jan.) and [467f1913](https://github.com/windmilleng/tilt/commit/467f1913ea08b562e2e16b26d8d7e458b92c941b) (Feb.) **print `docker push` and `docker pull` progress (respectively) to the Web UI**, so you're not in the dark about where your development time is going.
+
+Tilt tries to make microservice development as fast as possible, but sometimes there's just stuff you've gotta wait around for. However, when you inevitably have to wait around for something, you should know exactly what what you're waiting for and how it's progressing (and that your process isn't just hung somewhere and FUBAR).
```
Comment (ellenkorbes) [inline blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: `s/what what/what`

Comment (ellenkorbes) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: There's a high chance non-Americans won't know what FUBAR means.

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+These commits print `docker push / pull` progress output, updating lines in the Web UI in place as status meters creep forward. Users now have real-time visibility into what Docker is doing, instead of staring at a blank screen and hoping things are still working.
+
+![See docker push status in real time](/assets/images/jan-feb-2020-commit-of-the-month/push-output.gif)
```
Comment (ellenkorbes) [inline blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: Wouldn't it be better to inline this and save users a couple clicks?

Comment (maiamcc) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: Oh, this is markdown syntax for embedding an email -- did you think this was a clickable url?

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md (hunk 1 above)
Comment (maiamcc) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: ha whoops, I missed it b/c it didn't have "cotm" in the title. Will fix.

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md (hunk 4 above)
Comment (ellenkorbes) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: I did—missed the `!` there. My bad!

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md (hunk 2 above)
Comment (maiamcc) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: @nicks @hyu I don't want to mis-summarize and I'm not sure I'm seeing a good tie-in here. If you want to propose a few lines about this and where I should put them in, I totally will do!

Comment (nicks) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: Sure, how about:

We admire how much the Docker team invested in a great command-line user experience.

We've all felt the frustration of a command-line tool that sits their churning with no output. What is it doing? Will it ever finish? 

A good command-line tool at least gives you a simple progressbar or loading animation.

The Docker CLI goes way beyond that. It updates your terminal in-place with how big your layers are, whether they were cached, how much time each one is taking. It's responsive to your terminal width, summarizing the information differently depending on how much screen real estate it has to work with.

We wanted our web UI to do that progress display justice.

@hyu anything to add?

File: blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md (hunk 3 above)
Comment (maiamcc) [reply blog/_posts/2020-03-04-jan-feb-2020-commit-of-the-month.md]: fair, changed

**Pull request merged**